src/main/java/org/veupathdb/service/eda/ss/model/reducer/BinaryValuesStreamer.java

```diff
@@ -44,6 +47,26 @@ public <V, T extends VariableWithValues<V>> FilteredValueIterator<V, Long> strea
         VariableValueIdPair::getIdIndex);
   }
 
+  /**
+   * Stream entity IDs pulled based on the union or intersection of multiple streams of ID indexes from different
+   * string set filters.
+   * @param filter Filter to apply to variable values.
+   * @param study Study that the request is applicable to.
+   * @return
+   * @throws IOException
+   */
+  public Iterator<Long> streamMultiFilteredValues(
+      MultiFilter filter, Study study) throws IOException {
+    List<Iterator<Long>> idStreams = filter.getSubFilters().stream()
+        .map(Functions.fSwallow(subFilter -> streamFilteredValues(filter.getFilter(subFilter), study)))
+        .collect(Collectors.toList());
+    if (filter.getOperation() == MultiFilter.MultiFilterOperation.UNION) {
+      return new StreamUnionMerger(idStreams); // Intersect depending on operation.
+    } else { // operation == MultiFilter.MultiFilterOperation.UNION.INTERSECT
+      return new StreamIntersectMerger(idStreams);
+    }
+  }
+
   /**
    * Streams tuples of all entity ID indexes and the variable values associate with the variable passed in.
```

Comment:
i think this comment could clarify that we are emitting the string version of the value

Reply:
Yeah, good point. Fixed